### PR TITLE
Fix edge cases such as disable istio-inject for Tekton compiler

### LINKED ISSUE #114: Docs for deploying tekton pipeline on kfp user namespaces.
/kind feature

**Description:**
[A clear and concise description of what your proposal. What problem does it solve?]

We need step by step docs for users who want to deploy tekton on kfp namespace. (such as workaround on istio-inject without [kfp api](https://github.com/kubeflow/pipelines/blob/master/backend/src/apiserver/resource/resource_manager.go#L332-L340))

/assign

**Additional information:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Resolves #114 

**Description of your changes:**
- Add new annotation to disable istio-inject
- Remove literal string conversion for pyyaml since there could be conflicts with yaml 1.1 and 1.2 on some [edge cases](https://github.com/yaml/pyyaml/issues/98). If we want this feature back then we have to use ruamel.yaml. We are using pyyaml right now because we need to be consistent with the kfp package. 
- Add headers to golden test yaml
- Fix minor syntax bugs

**Environment tested:**

* Python Version (use `python --version`): 3.6.4
* Tekton Version (use `tkn version`): 0.11.3
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

### CODE REVIEW

File: sdk/python/tests/compiler/compiler_tests.py [hunk 1]
```diff
@@ -236,6 +237,23 @@ def _test_pipeline_workflow(self,
         compiled = list(yaml.safe_load_all(f))
       if GENERATE_GOLDEN_YAML:
         with open(golden_yaml_file, 'w') as f:
+          f.write(textwrap.dedent("""\
```
Comment (ckadner) [inline sdk/python/tests/compiler/compiler_tests.py:240]: can you move this text into a variable at the top of the file

File: sdk/python/kfp_tekton/compiler/compiler.py [hunk 2]
```diff
@@ -36,18 +36,6 @@
 from .. import tekton_api_version
 
 
-def _literal_str_representer(dumper, data):
```
Comment (ckadner) [inline sdk/python/kfp_tekton/compiler/compiler.py:39]: I forgot the history of this, can you quickly explain why we had to add this and why it's no longer needed?

File: sdk/python/kfp_tekton/compiler/_op_to_template.py [hunk 3]
```diff
@@ -445,17 +432,6 @@ def _op_to_template(op: BaseOp, enable_artifacts=False):
     if processed_op.init_containers:
         template['spec']['steps'] = _prepend_steps(processed_op.init_containers, template['spec']['steps'])
 
-    # initial base_step and volume setup
-    base_step = {
-        'image': 'busybox',
-        'name': 'copy-results',
-        'script': '#!/bin/sh\nset -exo pipefail\n'
-    }
-    volume_mount_step_template = []
```
Comment (ckadner) [inline sdk/python/kfp_tekton/compiler/_op_to_template.py:454]: aren't those still being used in the code below?

File: sdk/python/kfp_tekton/compiler/_op_to_template.py [hunk 4]
```diff
@@ -445,17 +432,6 @@ def _op_to_template(op: BaseOp, enable_artifacts=False):
     if processed_op.init_containers:
         template['spec']['steps'] = _prepend_steps(processed_op.init_containers, template['spec']['steps'])
 
-    # initial base_step and volume setup
-    base_step = {
-        'image': 'busybox',
-        'name': 'copy-results',
```
Comment (ckadner) [inline sdk/python/kfp_tekton/compiler/_op_to_template.py:451]: I still see the `copy-results` step in the compiled YAML and in a comment, does that come from somewhere else?

Comment (Tomcli) [reply sdk/python/kfp_tekton/compiler/_op_to_template.py:451]: I moved this into a different function, but somehow it get put back after the merge. We no longer need this over here because it is initiated at the beginning of this function.

File: sdk/python/kfp_tekton/compiler/compiler.py (hunk 2 above)
Comment (Tomcli) [reply sdk/python/kfp_tekton/compiler/compiler.py:39]: I wanted to add literal style to make the script part looks nicer. But there's some edge cases with pyyaml when using it on integer since it's designed with yaml 1.1

Here's the definition of literal style in yaml
https://yaml.org/spec/1.2/spec.html#id2795688

File: sdk/python/tests/compiler/compiler_tests.py (hunk 1 above)
Comment (Tomcli) [reply sdk/python/tests/compiler/compiler_tests.py]: done

File: sdk/python/kfp_tekton/compiler/_op_to_template.py (hunk 4 above)
Comment (ckadner) [reply sdk/python/kfp_tekton/compiler/_op_to_template.py:451]: right, thanks :-)

File: sdk/python/tests/compiler/compiler_tests.py [hunk 5]
```diff
@@ -226,18 +226,7 @@ def _test_pipeline_workflow(self,
...
-                                        generate_pipelinerun=generate_pipelinerun,
-                                        enable_artifacts=enable_artifacts)
-      with open(compiled_yaml_file, 'r') as f:
-        f = normalize_compiler_output_function(
-          f.read()) if normalize_compiler_output_function else f
-        compiled = list(yaml.safe_load_all(f))
-      if GENERATE_GOLDEN_YAML:
-        with open(golden_yaml_file, 'w') as f:
-          f.write(textwrap.dedent("""\
+    license_header = textwrap.dedent("""\
```
Comment (ckadner) [inline sdk/python/tests/compiler/compiler_tests.py:229]: I meant move `license_header` to the top of the `compiler_tests.py` file since Andrew will have to use that in his PR as well and it blows up the code too much here

Comment (Tomcli) [reply sdk/python/tests/compiler/compiler_tests.py:229]: oh you mean set it as global variable?

File: sdk/python/tests/compiler/compiler_tests.py [hunk 6]
```diff
@@ -27,6 +27,24 @@
 # in order to generate new "golden" YAML files
 GENERATE_GOLDEN_YAML = False
 
+# License header for Kubeflow project
+LICENSE_HEADER = textwrap.dedent("""\
```
Comment (ckadner) [inline sdk/python/tests/compiler/compiler_tests.py:31]: nice, thanks :-)
